kratos/includes/model_part.h

```diff
@@ -1716,35 +1716,47 @@ class KRATOS_API(KRATOS_CORE) ModelPart final
     ///@}
     ///@name Access
     ///@{
+    
 
+    /**
+     * @brief Access the @ref ProcessInfo related to the current @ref ModelPart tree.
+     */
     ProcessInfo& GetProcessInfo()
     {
-        return *mpProcessInfo;
+        return *this->pGetProcessInfo();
```

Comment:
the problem with this is that it generates a pointer even when it is not needed. This has a non negligible cost when you are in parallel

Reply:
Done! Addressed in commit fc762ca. Also made the same changes to the const GetProcessInfo().